Sources/Controls/UIBarButtonItem+Combine.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  UIBarButtonItem+Combine.swift
+//  CombineCocoa
+//
+//  Created by Shai Mishali on 12/08/2019.
+//  Copyright © 2019 Shai Mishali. All rights reserved.
+//
+
+import Combine
+import UIKit
+
+public extension UIBarButtonItem {
+  /// A publisher which emits whenever this UIBarButtonItem is tapped.
+  var tapPublisher: AnyPublisher<Void, Never> {
+    return TapPublisher(barButtonItem: self).eraseToAnyPublisher()
+  }
+}
+
+private extension UIBarButtonItem {
+  struct TapPublisher: Publisher {
```

Comment:
Any reason why you’re putting this inside `UIBarButtonItem` rather than inside `Combine.Publishers` like you did for `UIControl`?

Reply:
In this case it doesn't matter much because it's just a private extension on the UIBarButtonItem, it doesn't live outside of the scope of this file. The one other combine was publiers (e.g. `Combine.Publishers.ControlEvent`) so it was more important in that case. 

I don't have a huge preference to either direction, TBH.

Reply:
But the other could go under `UIControl` right?

I don’t mind - I just love consistency :-D

Reply:
I think there's a difference between public and private things, plus a bit different between things in a more generic scope (like a UIControl), and more specific scope (like a specific UI Component)

I see what Apple does is to put it on the component itself. I can align this for UIControl, but I'm not sure it matters too much in that more generic usability scope.